arangod/Aql/Condition.cpp

```diff
@@ -1073,8 +1074,9 @@ void Condition::collectOverlappingMembers(
 
       ::clearAttributeAccess(result);
 
+      TRI_ASSERT(!isFromTraverser) << "This is impossible!";
       if (rhs->isNullValue() &&
-          lhs->isAttributeAccessForVariable(result, isFromTraverser) &&
+          lhs->isAttributeAccessForVariable(result, false) &&
```

Comment:
It's clear why you changed it this way, but without the PR and the code alone, I think the reason for the assertion will elude a future reader. The error message isn't very meaningful, either. Just for readability, I would suggest two things:
* Leave `isFromTraverser` in (instead of changing it to false). It's obvious enough from the code that it's the same, and it will give a hint as to what the assertion is about.
* Give a more meaningful error message in the assertion; or delete it (we don't really expect to see it), but add a comment why this is important.

This is a cosmetic remark.